options/options_test.go

```diff
@@ -474,12 +474,18 @@ type optionsTester struct {
 }
 
 func createOptionsTestCases(s []string) []optionsTester {
-	return []optionsTester{
+	ret := []optionsTester{
 		{fmt.Sprintf("%s %s", s[0], s[2]), "mongodb://user:pass@foo", ShouldSucceed},
 		{fmt.Sprintf("%s %s", s[0], s[2]), fmt.Sprintf("mongodb://user:pass@foo/?%s=%s", s[1], s[2]), ShouldSucceed},
 		{fmt.Sprintf("%s %s", s[0], s[2]), fmt.Sprintf("mongodb://user:pass@foo/?%s=%s", s[1], s[3]), ShouldFail},
 		{"", fmt.Sprintf("mongodb://user:pass@foo/?%s=%s", s[1], s[2]), ShouldSucceed},
 	}
+	if s[0] == "--ssl" || s[0] == "--sslAllowInvalidCertificates" || s[0] == "--sslAllowInvalidHostnames" {
```

Comment:
Why have you done this?

Reply:
Because these params require no `value` part, in the test before it was set to `--ssl true`, `--sslAllowInvalidCertificates true`. This was not correct but worked before this change because `true` was parsed as a positional argument and ignored. With the change, we cannot allow unnecessary positional arguments.

Reply:
Oh I thought the go-flags library allowed passing --ssl=true or --ssl=false, but I'm totally wrong. Good change.

Reply:
Thanks!

Reply:
Thx!